core/src/banking_stage.rs

```diff
@@ -691,8 +687,7 @@ impl BankingStage {
 
     #[allow(clippy::too_many_arguments)]
     pub fn consume_buffered_packets(
```

Comment:
Behavior difference: this is only called *if* we have a valid bank to consume with now.

Reply:
Mmm, some subtle difference had already been introduced in `consume_buffered_packets()`. It [used to iterate through buffer by chunks, and check `poh_recorder` for working_bank _every_ iteration](https://github.com/solana-labs/solana/blob/v1.14.0/core/src/banking_stage.rs#L675-L696), so it exits from iterating though (and processing) the remaining buffer as soon as slot is expired. 

Right now, it applies the entire buffer to bank that may not longer process transactions. Would it be an issue?

Reply:
Potentially, with the MI change it became less efficient across slot boundaries, because we'd be exiting iteration early and then doing another full iteration.

It gives us a chance to receive new packets at the slot boundary (although we'd have a 0ms timeout if we have remaining packets).